From fb033217bfa7ca1b3b900ba448bd96688910b164 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 21 Jan 2025 18:30:31 +0800
Subject: [PATCH 6/8] loonggpu: gsgpu_vm: move struct definition to
 include/gsgpu_vm.h

Move struct definition to header so that non-static functions may access
them as parameters.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 gsgpu/gsgpu_vm.c   | 68 ----------------------------------------------
 include/gsgpu_vm.h | 68 ++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 68 insertions(+), 68 deletions(-)

diff --git a/gsgpu/gsgpu_vm.c b/gsgpu/gsgpu_vm.c
index 7859bf1..b369090 100644
--- a/gsgpu/gsgpu_vm.c
+++ b/gsgpu/gsgpu_vm.c
@@ -30,74 +30,6 @@
  * SI supports 16.
  */
 
-/**
- * struct gsgpu_pte_update_params - Local structure
- *
- * Encapsulate some VM table update parameters to reduce
- * the number of function parameters
- *
- */
-struct gsgpu_pte_update_params {
-
-	/**
-	 * @ldev: gsgpu device we do this update for
-	 */
-	struct gsgpu_device *ldev;
-
-	/**
-	 * @vm: optional gsgpu_vm we do this update for
-	 */
-	struct gsgpu_vm *vm;
-
-	/**
-	 * @src: address where to copy page table entries from
-	 */
-	u64 src;
-
-	/**
-	 * @ib: indirect buffer to fill with commands
-	 */
-	struct gsgpu_ib *ib;
-
-	/**
-	 * @func: Function which actually does the update
-	 */
-	void (*func)(struct gsgpu_pte_update_params *params,
-		     struct gsgpu_bo *bo, u64 pe,
-		     u64 addr, unsigned count, u32 incr,
-		     u64 flags);
-	/**
-	 * @pages_addr:
-	 *
-	 * DMA addresses to use for mapping, used during VM update by CPU
-	 */
-	dma_addr_t *pages_addr;
-
-	/**
-	 * @kptr:
-	 *
-	 * Kernel pointer of PD/PT BO that needs to be updated,
-	 * used during VM update by CPU
-	 */
-	void *kptr;
-};
-
-/**
- * struct gsgpu_prt_cb - Helper to disable partial resident texture feature from a fence callback
- */
-struct gsgpu_prt_cb {
-
-	/**
-	 * @ldev: gsgpu device
-	 */
-	struct gsgpu_device *ldev;
-
-	/**
-	 * @cb: callback
-	 */
-	struct dma_fence_cb cb;
-};
-
 /**
  * gsgpu_vm_bo_base_init - Adds bo to the list of bos associated with the vm
  *
diff --git a/include/gsgpu_vm.h b/include/gsgpu_vm.h
index 4e7da19..e71570a 100644
--- a/include/gsgpu_vm.h
+++ b/include/gsgpu_vm.h
@@ -60,6 +60,74 @@ struct gsgpu_bo_list_entry;
 #define GSGPU_VM_CONTEXT_GFX 0
 #define GSGPU_VM_CONTEXT_COMPUTE 1
 
+/**
+ * struct gsgpu_pte_update_params - Local structure
+ *
+ * Encapsulate some VM table update parameters to reduce
+ * the number of function parameters
+ *
+ */
+struct gsgpu_pte_update_params {
+
+	/**
+	 * @ldev: gsgpu device we do this update for
+	 */
+	struct gsgpu_device *ldev;
+
+	/**
+	 * @vm: optional gsgpu_vm we do this update for
+	 */
+	struct gsgpu_vm *vm;
+
+	/**
+	 * @src: address where to copy page table entries from
+	 */
+	u64 src;
+
+	/**
+	 * @ib: indirect buffer to fill with commands
+	 */
+	struct gsgpu_ib *ib;
+
+	/**
+	 * @func: Function which actually does the update
+	 */
+	void (*func)(struct gsgpu_pte_update_params *params,
+		     struct gsgpu_bo *bo, u64 pe,
+		     u64 addr, unsigned count, u32 incr,
+		     u64 flags);
+	/**
+	 * @pages_addr:
+	 *
+	 * DMA addresses to use for mapping, used during VM update by CPU
+	 */
+	dma_addr_t *pages_addr;
+
+	/**
+	 * @kptr:
+	 *
+	 * Kernel pointer of PD/PT BO that needs to be updated,
+	 * used during VM update by CPU
+	 */
+	void *kptr;
+};
+
+/**
+ * struct gsgpu_prt_cb - Helper to disable partial resident texture feature from a fence callback
+ */
+struct gsgpu_prt_cb {
+
+	/**
+	 * @ldev: gsgpu device
+	 */
+	struct gsgpu_device *ldev;
+
+	/**
+	 * @cb: callback
+	 */
+	struct dma_fence_cb cb;
+};
+
 /* VMPT level enumerate, and the hiberachy is:
  * DIR0->DIR1->DIR2
  */
-- 
2.48.1

